### PR TITLE
Adding stream fields dialog

### LINKED ISSUE #329: Add field dialog for Streamed data
Add a field dialog for when a user adds a field with the stream type. this should allow entering the broker&topic, schema, and schema specific fields (pv name for f142, etc.)

I think this should be added to the nexus file?

### PR DESCRIPTION
**Pull request by rerpha**
### Issue

Closes #329 

### Description of work

Adds a stream dialog and adds the stream to the nexus file. This should work for ev42, f142, ns10 and some of hs00 but not all. I have created ticket #422 for this.

### Acceptance Criteria 

Output command works perfectly with filewriter in all cases. 

### UI tests

Still in progress but would like to see if i'm going in the right direction with this first. 

### Nominate for Group Code Review

- [ ] Nominate for code review

### CODE REVIEW

**Comment by rerpha** [comment]
Todo: add UI tests

**Comment by rerpha** [comment]
TODO: add fields for f142

File: nexus_constructor/stream_fields_widget.py [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+        stream_group.create_dataset(
+            name="writer_module",
+            dtype=string_dtype,
+            data=self.schema_combo.currentText(),
+        )
+
+        schema = self.schema_combo.currentText()
+
+        if schema == "f142":
+            stream_group.create_dataset(
```
Comment (matthew-d-jones) [inline nexus_constructor/stream_fields_widget.py]: Extracting the contents of the `if` to a separate method would help keep things neat and readable here.

**Comment by rerpha** [comment]
marking as in-progress while i sort out the JSON output.

**Comment by rerpha** [comment]
Ok, there are a few things missing with this ticket. The writer is treating the streams as normal datasets because i have not given it a list of the streams in the first place. I also need to give the stream group an `NX_class` of "NXstream" or something similar. This could be covered pretty easily with a UI test too. Just writing this down here so I don't forget.

**Comment by rerpha** [comment]
Ready for another review.

**Pull request merged**